### PR TITLE
Fixed some Interpreter::TODOs

### PR DESCRIPTION
**Pull request by svyatonik**

### CODE REVIEW

**Comment by svyatonik** [comment]
Forgot to pass table element type to `TableType::new()`. Will add fix in a minute

**Comment by svyatonik** [comment]
Looks like there's no need in this currently :)

File: src/interpreter/runner.rs [hunk 1]
```diff
@@ -159,8 +159,9 @@ impl Interpreter {
 			body_stack.push_back(instruction_body);
 		}
 
+		let non_empty_stack_reason: &'static str = "body_stack contains entry for every frame_stack; frame_stack is not empty; qed";
```
Comment (NikVolf) [inline src/interpreter/runner.rs]: `:&' static str` is redundant, string literals are static

File: src/interpreter/memory.rs [hunk 2]
```diff
@@ -85,9 +87,8 @@ impl MemoryInstance {
 	pub fn grow(&self, pages: u32) -> Result<u32, Error> {
 		let mut buffer = self.buffer.write();
 		let old_size = buffer.len() as u32;
-		match pages.checked_mul(LINEAR_MEMORY_PAGE_SIZE).and_then(|bytes| old_size.checked_add(bytes)) {
+		match calculate_memory_size(old_size, pages, self.maximum_size) {
 			None => Ok(u32::MAX),
-			Some(new_size) if new_size > self.maximum_size => Ok(u32::MAX),
 			Some(new_size) => {
 				buffer.extend(vec![0; (new_size - old_size) as usize]);
```
Comment (NikVolf) [inline src/interpreter/memory.rs]: this is going to be double allocation for grown size
better to force buffer to allocate once and then fill it with zeros without allocating another zero-buffer
or leave another todo)

Comment (svyatonik) [inline src/interpreter/memory.rs]: Thanks for noticing :) replaced with `resize`

**Pull request merged**